Микросетки на флексах для всех страниц

ParovatkinA opened the pull request:
---
:mortar_board: [Микросетки на флексах для всех страниц](https://up.htmlacademy.ru/htmlcss/33/user/302695/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/302695-sedona-33/7/

ParovatkinA committed:
Добавляет правила для селектора visually-hidden

ParovatkinA committed:
Отключает стандартные буллеты из навигации

ParovatkinA committed:
задает ссылкам блочную модель

ParovatkinA committed:
увеличивает область клика пользовательского меню

ParovatkinA committed:
добавляет отступы от логотипа

чтобы шапка и контент не "прилипал" к нему

ParovatkinA committed:
добавляет общие стили navigation-list для пользовательского меню

ParovatkinA committed:
добавляет внешние отступы для блока first-screen и advantages

ParovatkinA committed:
отключает булеты у списка преимуществ

ParovatkinA committed:
добавляет стилизацию трем элементам блока преимущества

ParovatkinA committed:
добавляет стили в блок first-screen

ParovatkinA committed:
добавляет отступы в footer

ParovatkinA committed:
добавляет разметку к хлебным крошкам

ParovatkinA committed:
добавляет базовую стилизацию к пагинации

ParovatkinA committed:
добавляет и изменяет стили в разделе "сортировка" и "подписка"

ParovatkinA committed:
исправляет код на главной странице

добавляет обертку всем блокам главной страницы, добавляет div для логотипа, перемещает секцию first-screen в преимущества, изменяет список преимуществ и пр.

ParovatkinA committed:
исправляет код на странице каталога

обертка всего содержимого в div, создает для формы фильтра выбора отеля отдельный div и пр.

ParovatkinA committed:
удаляет <br>

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
+  text-align: center;
+}
+
+.advantages-list-2 {
```
при добавлении еще одного элемента, появляется горизонтльный скролл
![image](https://user-images.githubusercontent.com/41639106/154730676-0e6fd0bb-e0b8-432b-a802-b72335340746.png)
а новый блок должен вставать на новую строку. по-умолчанию флекс выставляет все элементы в одну строку, и для того что бы элементы, при переполнении перемещались на новую строку, необходимо флекс элементу разрешить этот перенос

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
+
+.advantages-list-2 {
+  display: flex;
+  width: 400px;
```
ширина у этого блока больше чем 400рх, и, содержимое выходит за пределы этого блока. ширину можно либо совсем убрать(лучше), либо заменить на нужную

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
+background: #83B3D333;
+}
+
+.bester-reason:nth-child(3) {
```
псевдо-классы можно, и, даже нужно использовать. но в таком виде они используются редко. гораздо удобнее указывать не на конкретный элемент, а использовать формулы или ключевые слова. В данном случае можно всем не четным элементам задать серый фон. При таком походе добавляя еще элементы(сколько угодно) все четные элементы будут окрашиваться серым фоном
https://developer.mozilla.org/ru/docs/Web/CSS/:nth-child

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
-.advantages-item:nth-child(4) {
+  display: flex;
+  flex-wrap: wrap;
+  width: 400px;
```
нужно проверить по всему макету соответствие размеров. и у этого блока ширина не по макету

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
+  text-transform: uppercase;
+}
+
+.advantages-title-3 {
```
![image](https://user-images.githubusercontent.com/41639106/154733371-b97713f3-2710-44c5-bf91-d3a58d916bbd.png)
по всему макету нужно проверить: у многих элементов есть по-умолчанию некоторые свойства, которые нужно отменять/изменять, в данном случае это внешние отступы сверху и снизу.

uzinok commented on index.html:
```diff
@@ -16,14 +16,13 @@
         </li>
         <li class="navigation-item">
-          <a class="navigation-link" href="index.html#about-sedona">О Седоне</a>
+          <a class="navigation-link" href="index-02.html#about-sedona">О Седоне</a>
```
ссылки по всему проекту должны либо иметь заглушку `href="#"` (если нет страницы в макете), либо вести на страницу "заглушку". Это страница с произвольной версткой, внутри которой есть ссылки на главную страницу, и на страницу каталога

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
-    text-align: center;
-  }
+  background-color: #ffffff;
+  background-image: url("../images/index-background.jpg");
```
этот фон должен быть у блока `.logo-welcome`

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
+  text-transform: uppercase;
+}
 
 .button-find {
```
![image](https://user-images.githubusercontent.com/41639106/154735457-c00b9d44-3a50-450d-aafd-242bace0d3c1.png)
этому классу нужно задавать только те правила, которые добавляются/изменяются, в отличии от основного класса. перезаписывать одни и те же стили нет смысла. 
Еще есть следующая причина по которой не стоит перезаписывать эти правила:
при изменении размера шрифта дизайнером - изменится размер шрифта у всех кнопок, и его, сейчас, нужно будет изменять во многих местах кода
Проверь весь код

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,825 @@
+  line-height: 36px;
+}
 
 .find-hotel p {
```
![image](https://user-images.githubusercontent.com/41639106/154737749-84e9d4fd-655f-4ee4-9e32-3bf3f8f50f6a.png)
этот блок не стоит ограничивать по ширине, и устанавливать по центру кнопку с помощью внутреннего отступа.
кнопка, а точнее ссылка - строковый элемент, ее поставить по-центру можно только одним выравниванием текста.

ParovatkinA committed:
исправляет название ссылки в навигации

вместо "Главное меню" - на "Главная"

ParovatkinA committed:
удаляет свойства селектора .main-page и добавляет их селектору .logo-welcome

ParovatkinA committed:
удаляет ненужный класс main-page

ParovatkinA committed:
удаляет отступы сверху и снизу в логотипе главной страницы и размещает по центру текст в .advantages

ParovatkinA committed:
изменяет отступы в <h2> и <p> блока first-screen

ParovatkinA committed:
добавляет отступы в секцию find-hotel

ParovatkinA committed:
изменяет стили для вложенного списка причины №1

убирает ширину 400px для класса advantages-item, добавляет свойство flex-wrap: wrap для вложенного списка advantages-list-2

ParovatkinA committed:
добавляет фон для каждого преимущества и удаляет ненужные отступы

ParovatkinA committed:
добавляет стили блоку inner-header-filter и делает inner-find-hotel флексовым элементом

ParovatkinA committed:
делает блок с фильтром гостиниц на флексах

ParovatkinA committed:
добавляет отступы блоку с карточками find-hotel-catalog

ParovatkinA committed:
меняет маркированный список пагинации на нумерованный

ParovatkinA committed:
добавляет отступы в footer

ParovatkinA committed:
добавляет aria-current="page" текущей ссылке

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,813 @@
+  background: #83B3D31F;
+}
+
+.three-reason:nth-child(3n+2) {
```
![image](https://user-images.githubusercontent.com/41639106/154999552-84830328-17ad-4c1d-a005-928800aa0c49.png)
в данном случае эту формулу лучше заменить, и белый фон задавать четным элементам
![image](https://user-images.githubusercontent.com/41639106/154999754-cc3676df-a4f4-4a63-95c1-fa1143be2d89.png)

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,813 @@
-.advantages-item:nth-child(8) {
-    background: #83B3D31F;
-  }
+.best-reason-2:nth-child(3n+4) {
```
формула `.best-reason-2:nth-child(3n+4)` выбирает каждый третий элемент начиная с четвертого, а нам нужно каждый второй
`3n` - выбирает каждый третий
`+ 4` - начиная с четвертого элемента
например `4n` - будет выбирать каждый четвертый

uzinok commented on styles/styles.css:
```diff
@@ -1,482 +1,813 @@
+  background-repeat: no-repeat;
+}
+
+.newsletter {
```
![image](https://user-images.githubusercontent.com/41639106/155000732-1f3eb75d-729d-49d4-911f-75dd5de694dd.png)
много стилей повторяются

uzinok commented on catalog.html:
```diff
@@ -232,8 +236,7 @@ <h3 class="hotel-card-title">GreenTree Inn</h3>
               <li class="pagination-item">
```
этому элементу необходимо добавить атрибут `value` со значением `10`. В таком случае скрин-ридер прочитает этот элемент как десятый по счету

uzinok reviewed (commented):
в целом хорошо, но не без правок... многое поняла, что меня очень радует.

ParovatkinA committed:
удаляет повторяющиеся правила у кнопки с классом .button-find

ParovatkinA committed:
удаляет повторяющиеся правила

ParovatkinA committed:
добавляет <br> для переноса текста

вынужденная мера - так как рекомендовано убрать ширину у данного блока и удалить внутренние отступы

ParovatkinA committed:
удаляет лишние правила и стили в css

ParovatkinA committed:
удаляет лишние стили в newsletter

ParovatkinA committed:
изменяет стили в блоке "поиск гостиницы"

ParovatkinA committed:
корректирует стили и отступы всех страниц
